tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionStringTests.cs

```diff
@@ -296,6 +300,16 @@ public void When_everything_is_specified()
             subject.AuthMechanismProperties.Count.Should().Be(1);
             subject.AuthMechanismProperties["canonicalize_host_name"].Should().Be("true");
             subject.AuthSource.Should().Be("admin");
+            subject
+                .Compressors
+                .Should()
+#if NET452 || NETSTANDARD2_0
+                .Contain(x => x.Type == CompressorType.Snappy)
+                .And
+#endif
+                .Contain(x => x.Type == CompressorType.Zlib)
+                    .And
+                    .Contain(x => x.Properties.ContainsKey("Level"));
```

Comment:
I find this complex logical one liner difficult to understand, made even more so by the fact that part of this one expression is conditionally compiled. After much study I concluded that this is what is being tested:

    #if NET452 || NETSTANDARD2_0
            var expectedCompressorTypes = new[] { CompressorType.Snappy, CompressorType.Zlib };
    #else
            var expectedCompressorTypes = new[] { CompressorType.Zlib };
    #endif
            subject.Compressors.Select(x => x.Type).Should().Equal(expectedCompressorTypes);
            subject.Compressors.Single(x => x.Type == CompressorType.Zlib).Properties["Level"].Should().Be(4);

Would you consider refactoring to this?